### PR TITLE
review how the compile command works

### PR DESCRIPTION
**Pull request by lyrixx**
* we can now build castor, and the examples app
* it's really static, example with an alpine container
    image

### CODE REVIEW

File: doc/going-further/compile.md [hunk 1]
```diff
@@ -1,21 +1,22 @@
-# Compiling your application into a standalone binary
+# Compiling your application into a static binary
```
Comment (lyrixx) [inline doc/going-further/compile.md:1]: I tested, and it's really static 🎉

File: examples/signal.php [hunk 2]
```diff
@@ -4,6 +4,10 @@
 
 use Castor\Attribute\AsTask;
 
+if (!\defined('SIGUSR2')) {
+    \define('SIGUSR2', 12);
```
Comment (lyrixx) [inline examples/signal.php:8]: not defined in the static binary with the defaults options `mbstring,phar,posix,tokenizer`

File: src/Console/Command/CompileCommand.php [hunk 3]
```diff
@@ -30,8 +31,9 @@ protected function configure(): void
     {
         $this
             ->setName('compile')
+            ->setAliases(['castor:compile'])
```
Comment (lyrixx) [inline src/Console/Command/CompileCommand.php:34]: in case one redefine the command name, same for repack

File: src/FunctionFinder.php [hunk 4]
```diff
@@ -69,6 +69,11 @@ private function doFindFunctions(iterable $files): iterable
             yield from $this->resolveListeners($reflectionFunction);
         }
 
+        // Symfony task in repacked application are not supported
```
Comment (lyrixx) [inline src/FunctionFinder.php:72]: let's be lazy for now. No time to dig this part...

File: doc/going-further/compile.md [hunk 5]
```diff
@@ -39,15 +40,21 @@ vendor/bin/castor compile --help
...
+   phar into a binary.
+2. If required, it automatically installs dependencies and compiles PHP with the
+   specified extensions.
 3. Combines the compiled PHP and your phar file into a single executable.
 
 ## Post-Compilation
 
-Once the compilation is finished, your Castor application is transformed into a standalone binary named `castor` by default (you can use the `--output` option to change it). 
+Once the compilation is finished, your Castor application is transformed into a
+static binary named `castor` by default (you can use the `--output` option to
```
Comment (pyrech) [inline doc/going-further/compile.md:53]: Isn't the option renamed --binary-path?

File: src/Console/Command/RepackCommand.php [hunk 6]
```diff
@@ -20,6 +20,7 @@ protected function configure(): void
     {
         $this
             ->setName('repack')
+            ->setAliases(['castor:repack'])
```
Comment (pyrech) [inline src/Console/Command/RepackCommand.php:23]: Should'nt we inverse the name and alias?

File: doc/going-further/compile.md (hunk 5 above)
Comment (lyrixx) [reply doc/going-further/compile.md]: fixed

File: src/Console/Command/RepackCommand.php (hunk 6 above)
Comment (lyrixx) [reply src/Console/Command/RepackCommand.php]: you are right, and it's now consistant with the debug command

**Review by pyrech — APPROVED** [review]

**Pull request merged**